TestDefaultValuesAreSettable fails after TestTiDBAutoAnalyzeConcurrencyValidation/Auto_analyze_disabled

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```bash
(cd pkg/sessionctx/variable ; go test -v -tags intest -failfast -run "TestDefaultValuesAreSettable|TestTiDBAutoAnalyzeC/Auto_analyze_di")
```

### 2. What did you expect to see? (Required)
test passes

### 3. What did you see instead (Required)
```
=== RUN   TestTiDBAutoAnalyzeConcurrencyValidation
=== RUN   TestTiDBAutoAnalyzeConcurrencyValidation/Auto_analyze_disabled
--- PASS: TestTiDBAutoAnalyzeConcurrencyValidation (0.00s)
    --- PASS: TestTiDBAutoAnalyzeConcurrencyValidation/Auto_analyze_disabled (0.00s)
=== RUN   TestDefaultValuesAreSettable
    variable_test.go:454: 
        	Error Trace:	/Users/mattias/repos/worktrees/tidb-main/pkg/sessionctx/variable/variable_test.go:454
        	Error:      	Received unexpected error:
        	            	cannot set tidb_auto_analyze_concurrency: requires both tidb_enable_auto_analyze and tidb_enable_auto_analyze_priority_queue to be true. Current values: tidb_enable_auto_analyze=false, tidb_enable_auto_analyze_priority_queue=true
        	            	github.com/pingcap/tidb/pkg/sessionctx/variable.init.func333
        	            		/Users/mattias/repos/worktrees/tidb-main/pkg/sessionctx/variable/sysvar.go:1662
        	            	github.com/pingcap/tidb/pkg/sessionctx/variable.(*SysVar).Validate
        	            		/Users/mattias/repos/worktrees/tidb-main/pkg/sessionctx/variable/variable.go:232
        	            	github.com/pingcap/tidb/pkg/sessionctx/variable.TestDefaultValuesAreSettable
        	            		/Users/mattias/repos/worktrees/tidb-main/pkg/sessionctx/variable/variable_test.go:453
        	            	testing.tRunner
        	            		/usr/local/go/src/testing/testing.go:1792
        	            	runtime.goexit
        	            		/usr/local/go/src/runtime/asm_arm64.s:1223
        	Test:       	TestDefaultValuesAreSettable
--- FAIL: TestDefaultValuesAreSettable (0.00s)
```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
current master: 6eeb23e31753a026b1d241f2de7db0623518e35b

--- a/pkg/sessionctx/variable/sysvar_test.go
+++ b/pkg/sessionctx/variable/sysvar_test.go
@@ -1877,13 +1877,6 @@ func TestTiDBAutoAnalyzeConcurrencyValidation(t *testing.T) {
 		expectError         bool
 	}{
 		{
-			name:                "Both enabled, valid input",
-			autoAnalyze:         true,
-			autoAnalyzePriority: true,
-			input:               "10",
-			expectError:         false,
-		},
-		{
 			name:                "Auto analyze disabled",
 			autoAnalyze:         false,
 			autoAnalyzePriority: true,
@@ -1903,6 +1896,14 @@ func TestTiDBAutoAnalyzeConcurrencyValidation(t *testing.T) {
 			autoAnalyzePriority: false,
 			input:               "10",
 			expectError:         true,
+		},
+		// Last so it ends as its defaults
+		{
+			name:                "Both enabled, valid input",
+			autoAnalyze:         true,
+			autoAnalyzePriority: true,
+			input:               "10",
+			expectError:         false,
 		},
 	}
 

